### PR TITLE
Sg/leios ib generation

### LINKED ISSUE #43: Simulate Simple Leios IB production in rust
- Parameterize rate of input block production (remember 𝑓I can be >1, and if it is then we need to run run multiple lotteries)
- Simulate gossip for headers, FFD dissemination for bodies

### PR DESCRIPTION
**Pull request by SupernaviX**
Fixes #43

### CODE REVIEW

**Review by abailly-iohk — COMMENTED** [review]
I tried to run the simulation based on the `README` instructions and it did not compile:

```
Error: TOML parse error at line 1, column 1
  |
1 | block_generation_probability = 0.05
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
missing field `ib_generation_probability`
```

**Comment by SupernaviX** [comment]
> I tried to run the simulation based on the `README` instructions and it did not compile:
> 
> ```
> Error: TOML parse error at line 1, column 1
>   |
> 1 | block_generation_probability = 0.05
>   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> missing field `ib_generation_probability`
> ```

I forgot to update test parameters for the bigger file😅 pushed that now.

File: sim-rs/src/model.rs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+id_wrapper!(InputBlockId, u64);
+
+#[derive(Debug, Clone, Hash, PartialEq, Eq)]
+pub struct InputBlockHeader {
+    pub slot: u64,
+    pub producer: NodeId,
+    /// Need this field to distinguish IBs from the same slot+producer.
+    /// The real implementation can use the VRF proof for that.
```
[the remark this answers is not in the document]
Comment (Quantumplation) [reply sim-rs/src/model.rs:51]: I think this is related to [this conversation](https://input-output-rnd.slack.com/archives/C074AHSKJF7/p1728671819463119). If the 𝑓I parameter is greater than 1, multiple IBs can be produced in the same slot.

[the remark this answers is not in the document]
Comment (SupernaviX) [reply sim-rs/src/model.rs:51]: @abailly-iohk this field is conceptually an index into the slot, you can think of it as a subslot index. If a node is allowed to produce two IBs in the same slot, one will have index 0 and one will have index 1. 

I haven't really thought about equivocation much at all yet, but I kind of assumed that `slot` was meant to include the subslot in that paper. It doesn't make much sense to me for the protocol to allow a node to produce several IBs, but then penalize them for doing so.

File: sim-rs/src/model.rs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    pub index: u64,
+    pub timestamp: Timestamp,
+}
+impl InputBlockHeader {
+    pub fn id(&self) -> InputBlockId {
+        let mut hasher = DefaultHasher::new();
+        (self.slot, self.producer, self.index).hash(&mut hasher);
```
[the remark this answers is not in the document]
Comment (SupernaviX) [reply sim-rs/src/model.rs:58]: Yeah, I'll get rid of that. I wanted smaller keys and didn't think about it as hard as I should have

**Pull request merged**